packages/material-ui-lab/src/SpeedDialAction/SpeedDialAction.js

```diff
@@ -78,22 +85,55 @@ class SpeedDialAction extends React.Component {
       ...other
     } = this.props;
 
-    let clickProp = { onClick };
+    let clickProps = { onClick };
     if (typeof document !== 'undefined' && 'ontouchstart' in document.documentElement) {
       let startTime;
-      clickProp = {
+      clickProps = {
+        ...clickProps,
         onTouchStart: () => {
           startTime = new Date();
         },
-        onTouchEnd: () => {
+        onTouchEnd: event => {
           // only perform action if the touch is a tap, i.e. not long press
           if (new Date() - startTime < 500) {
             onClick();
           }
+          event.preventDefault();
+          event.stopPropagation();
         },
       };
     }
 
+    clickProps.style = { transitionDelay: `${delay}ms` };
+
+    const actionButton = (
+      <Button
+        variant="fab"
+        mini
+        className={classNames(classes.actionButton, classes.button, !open && classes.buttonClosed)}
+        tabIndex={-1}
+        role="menuitem"
+        {...ButtonProps}
+        {...clickProps}
```

Comment:
What happens to the event handlers that are passed by the user? We use https://github.com/mui-org/material-ui/blob/ffe7bd0071eb78ea81cc78027bd34f91d2e7896f/packages/material-ui/src/utils/helpers.js#L47 throughout our codebase to avoid these issues.